### ISSUE TITLE
Floating-point non-determinism in GPU operations

### ISSUE THREAD

**Issue by isaac-chung** [embeddings-benchmark/mteb#2789]
Wherever we import torch. [Reference](https://discuss.pytorch.org/t/effect-of-torch-backends-cudnn-deterministic-true/113095).

```python

import torch
 
torch.manual_seed(seed)
torch.backends.cudnn.deterministic = True
torch.backends.cudnn.benchmark = False

```

**Comment by Samoed** [comment]
I think [`transformers.set_seed`](https://github.com/huggingface/transformers/blob/f4fc42216cd56ab6b68270bf80d811614d8d59e4/src/transformers/trainer_utils.py#L93) cover this case

**Comment by KennethEnevoldsen** [comment]
Hmm, how does this work if someone sets the random seed?

Could I e.g. seed hack the benchmark using:
```
import mteb

# set seed
```

Is there anywhere in the repo where we use random number generators from torch? (I only see a few places in tests)

There are ways to prevent this in numpy, some of them, as well as arguments for not using global seeds, are mentioned [here](https://builtin.com/data-science/numpy-random-seed)

**Comment by Samoed** [comment]
> Is there anywhere in the repo where we use random number generators from torch?

I don't think so, but in `v2` I've added `transformers.set_seed` too https://github.com/embeddings-benchmark/mteb/blob/9c63639f76f45bd64ace7ccc34809ce40648e115/mteb/abstasks/AbsTask.py#L30

**Comment by KennethEnevoldsen** [comment]
So we can close this?

**Comment by isaac-chung** [comment]
Added to v2 issue: I'd like to use to remind us to actually test it out in v2, e.g. run 3-5 runs of the same small collection of representative datasets & models, if that's alright?

**Comment by KennethEnevoldsen** [comment]
Perfectly fine with me (we need some validation of v2 anyway)

**Comment by KennethEnevoldsen** [comment]
This essentially just needs to be tested by 
> run[ning] 3-5 runs of the same small collection of representative datasets & models
